### PR TITLE
Migrate Makefile to Magefile

### PR DESCRIPTION
**Pull request by rappizs**
Migrated `make` to `mage`

Fixes https://github.com/uwu-tools/gh-jira-issue-sync/issues/61

### CODE REVIEW

**Comment by lgecse** [comment]
LGTM! 🙌

**Review by justaugustus — APPROVED** [review]
Great improvement, @rappizs!

I've built on this PR a bit to integrate some patterns from some of the Kubernetes RelEng repos:
- https://github.com/kubernetes-sigs/bom/blob/main/magefile.go
- https://github.com/kubernetes-sigs/tejolote/blob/main/magefile.go

File: magefile.go [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+	// Environment variables.
+	envLDFLAGS      = "GHJIRA_LDFLAGS"
+	envMinGoVersion = "MIN_GO_VERSION"
+
+	// Test variables.
+	coverMode            = "atomic"
+	coverProfileFilename = "unit-coverage.out"
+)
+
+// All runs all targets for this repository
```
Comment (rappizs) [inline magefile.go:54]: The description is a little bit misleading, since we only run two targets not all of them

File: magefile.go [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+
+	for _, clean := range toClean {
+		sh.Rm(clean)
+	}
+
+	fmt.Println("Done.")
+}
+
+// getBuildDateTime gets the build date and time
+func getBuildDateTime() string {
```
Comment (rappizs) [inline magefile.go:262]: This func is not used anywhere

File: magefile.go [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+}
+
+// getCommit gets the hash of the current commit
+func getCommit() string {
+	commit, _ := sh.Output("git", "rev-parse", "--short", "HEAD")
+	return commit
+}
+
+// getGitState gets the state of the git repository
+func getGitState() string {
```
Comment (rappizs) [inline magefile.go:281]: This func is not used anywhere

File: magefile.go [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+	if err := BuildImages(); err != nil {
+		return fmt.Errorf("building the images: %w", err)
+	}
+
+	return nil
+}
+
+func Clean() {
+	fmt.Println("Cleaning workspace...")
+	toClean := []string{"output"}
```
Comment (rappizs) [inline magefile.go:252]: As far as I understand we don't generate anything to a folder called `output` so this Clean func is not cleaning the repo properly. Instead we should remove `bin`, as we did in the Makefile.

**Review by rappizs — COMMENTED** [review]
Thanks for the refactoring @justaugustus!

I found some minor issues with the final code, could you please take a look at them? Should we create an issue for it or is it ok if I just create a PR for fixing them?

File: scripts/verify-build.sh [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    linux/arm64
+    darwin/amd64
+    darwin/arm64
+)
+
+for PLATFORM in "${PLATFORMS[@]}"; do
+    OS="${PLATFORM%/*}"
+    ARCH=$(basename "$PLATFORM")
+
+    output_name=bom'-'$OS'-'$ARCH
```
Comment (rappizs) [inline scripts/verify-build.sh:34]: I believe `bom` is just a copy mistake from https://github.com/kubernetes-sigs/bom/blob/main/scripts/verify-build.sh, this should be gh-jira-issue-sync or something like that.

File: magefile.go [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+
+	if err := Build(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Build runs go build
+func Build() error {
```
Comment (rappizs) [inline magefile.go:126]: Build doesn't work on OSX, created an issue for it: https://github.com/uwu-tools/gh-jira-issue-sync/issues/166

File: scripts/verify-build.sh [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+    if [ "$OS" = "windows" ]; then
+        output_name+='.exe'
+    fi
+
+    echo "Building project for $PLATFORM"
+    CGO_ENABLED=0 GOARCH="$ARCH" GOOS="$OS" go build -trimpath -ldflags "${GHJIRA_LDFLAGS}" -o output/$output_name ./main.go
```
Comment (rappizs) [inline scripts/verify-build.sh:41]: The `output` dir is not included in the .gitignore file.

**Pull request merged**